packages/bridge/contracts/l1/L1GovernanceRelay.sol

```diff
@@ -21,6 +21,8 @@ import "arb-bridge-eth/contracts/bridge/interfaces/IInbox.sol";
 import "../l2/L2GovernanceRelay.sol";
 
 // Relay a message from L1 to L2GovernanceRelay
+// Sending L1->L2 message on arbitrum requires sending some ETH. That's why this contract can receive ether.
```

Comment:
You don't always require sending eth.
If the `from` address in the L2 holds funds, those can be used to pay for gas.
Note that in this case it would be the address `applyL1ToL2Alias(L1GovernanceRelay)` that would need to hold funds

Reply:
The L1 contract could then claim these funds back using something like this https://gist.github.com/fredlacs/7abd3569be1e1dc8cb2a702fdda663d8

Reply:
Clarified the comment thanks! Is there any advantage of sending eth on l2 rather than on l1? its easier for us to maintain it in one place. Also, do you think we should add reclaiming from L2? ie. when ETH could be refunded?